source/frontend/src/features/documents/documentDetail/DocumentDetailContainer.tsx

```diff
@@ -140,37 +231,43 @@ export const DocumentDetailContainer: React.FunctionComponent<
     props.pimsDocument.documentType?.mayanId,
     getDocumentTypeMetadataApiCall,
     isMounted,
+    props.relationshipType,
   ]);
 
-  const onUpdateDocument = async (updateRequest: ApiGen_Requests_DocumentUpdateRequest) => {
-    if (props.pimsDocument.id) {
-      const result = await updateDocument(props.pimsDocument.id, updateRequest);
-      result && props.onUpdateSuccess();
+  useEffect(() => {
+    if (isEditable) {
+      fetchDocumentTypes();
     }
-  };
+  }, [fetchDocumentTypes, isEditable]);
 
-  if (isEditable === false) {
```

Comment:
nit. Looks like this switched the internal component. Was that for readability? not an issue, but it makes the git diff harder to assess

Reply:
readability.  since 'isEditable' is already a boolean.